Sources/StreamChat/Models/Channel.swift

```diff
@@ -144,7 +144,14 @@ public struct ChatChannel {
     
     /// Additional data associated with the channel.
     public let extraData: [String: RawJSON]
-
+    
+    /// The channel message is supposed to be shown in channel preview.
+    ///
+    /// - Important: The `previewMessage` can differ from `latestMessages.first` (or even not be included into `latestMessages`)
+    /// because the preview message is the last `non-deleted` message sent to the channel.
+    public var previewMessage: ChatMessage? { _previewMessage }
+    @CoreDataLazy private var _previewMessage: ChatMessage?
```

Comment:
I'm wondering if we could avoid this 🤔 One of the main issues in the channel list right now is how we update the message in the channel list item. This sometimes blocks the main thread because of the way we fetch the `latestMessages.first` from the main thread and we use `performAndWait` in CoreData. Since we are moving from `latestMessages` to `previewMessage` it could be an opportunity to improve this 🤔 But not sure if this is possible right now without breaking changes

Reply:
Do you want it to be just `public var previewMessage: ChatMessage?` i.e. without core data lazy?

Reply:
Yes, although I'm not sure about the impact of this 🤔

Reply:
There'll be no impact. In both cases, DB request happens on main thread, the only different is **when** it happens. Without core data lazy, model evaluation will happen before the change is propagated to the UI. With lazy stuff, it's gonna happen when we're calculating `subtitle` for the channel cell.

Reply:
So probably is better that we keep the CoreDataLazy for now? Otherwise, the impact will move to the loading of the channels right?